torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1661,6 +1661,15 @@ class ShapePropagator {
       std::swap(strides.at(0), strides.at(1));
       node->output()->setType(tp->withSizesStrides(sizes, strides));
       return true;
+    } else if (node->matches("aten::linear(Tensor input, Tensor weight, Tensor? bias) -> Tensor")) {
+      auto input_type = tensor_types.at(0);
+      auto sizes = input_type->sizes();
+      auto weight_type = tensor_types.at(1);
+      int64_t last_dim = sizes.size() - 1;
+      SHAPE_ASSERT(weight_type->sizes().size() == 2 && sizes.size() >= 2);
+      sizes.at(last_dim) = weight_type->sizes()[0];
+      node->output()->setType(input_type->withSizes(sizes));
+      return true;
```

Comment:
I don't think we need to add complete shape prop, since we're not really using it at this point. Also, this looks like the same case as for `aten::mm`, so maybe we could avoid duplicating them?

Reply:
`aten::mm` section requires the input to be 2, where `aten::linear` have 3.. I think I can also do partial shape prop like the `aten::bilinear` defined [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/shape_analysis.cpp#L1470), but this also requires additional code (which is a bit duplicated)..

Reply:
Fair enough, let's leave it.